crates/accelerate/src/quantum_circuit/circuit_data.rs

```diff
@@ -300,6 +306,155 @@ impl CircuitData {
         self.data.reserve(additional);
     }
 
+    /// Returns a tuple of the sets of :class:`.Qubit` and :class:`.Clbit` instances
+    /// that appear in at least one instruction's bit lists.
+    pub fn active_bits(&self, py: Python<'_>) -> Py<PyTuple> {
+        let mut qubits: HashSet<BitType> = HashSet::new();
+        let mut clbits: HashSet<BitType> = HashSet::new();
```

Comment:
Is there a `with_capacity()` we can do here to save allocations. I'd be inclined to suggest using an iterator collection here but because that would result in two iterations I think just using `with_capacity()` here instead of `new()` could save us some time. I'm thinking if we just did `with_capacity(self.qubits_native.len())` this should be fine we'd potentially over allocate if there are a lot of idle qubits, but this is just temporary as we subsequently iterate over the list for the python output and the sets will be dropped as the function exits.

```suggestion
        let mut qubits: HashSet<BitType> = HashSet::with_capacity(self.qubits_native.len());
        let mut clbits: HashSet<BitType> = HashSet::with_capacity(self.clbits_native.len());
```

Reply:
Done in d42b4f74987f65f0c5afe09daf686ebf59f13bb9.